dask_jobqueue/core.py

```diff
@@ -219,29 +278,31 @@ def stop_workers(self, workers):
         """ Stop a list of workers"""
         if not workers:
             return
-        workers = list(map(int, workers))
-        jobs = [self.jobs[w] for w in workers]
+        jobs = {_job_id_from_worker_name(w.name) for w in workers}
         self._call([self.cancel_command] + list(jobs))
-        for w in workers:
-            with ignoring(KeyError):
-                del self.jobs[w]
 
     def scale_up(self, n, **kwargs):
         """ Brings total worker count up to ``n`` """
-        return self.start_workers(n - len(self.jobs))
+        pending_workers = self.worker_processes * len(self.pending_jobs)
+        active_and_pending = len(self.scheduler.workers) + pending_workers
```

Comment:
A probably rare case, but you may miss starting workers here, when a job just began to start, so is moving from pending to running. We may have some worker process started for a given job_id, but not all of them.
Maybe it is safer to just rely on self.pending_jobs and self.running_jobs, but I'm not sure, we could also miss ending jobs 🙂 ...